recipes/gtk4/meta.yaml

```diff
@@ -0,0 +1,138 @@
+{% set name = "gtk" %}
+{% set version = "4.0.3" %}
+{% set version_majmin = '.'.join(version.split('.', 2)[:2]) %}
+
+package:
+  name: gtk4  # note customization
+  version: {{ version }}
+
+source:
+  url: https://download.gnome.org/sources/{{ name }}/{{ version_majmin }}/{{ name }}-{{ version }}.tar.xz
+  sha256: d7c9893725790b50bd9a3bb278856d9d543b44b6b9b951d7b60e7bdecc131890
+  patches:
+    # https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/3342
+    - gdkmacosglcontext.c.patch  # [osx]
+
+build:
+  number: 0
+  run_exports:
+    - {{ pin_subpackage('gtk4', max_pin='x') }}
+    # gtk apps need at least a default icon set for fallback
+    - adwaita-icon-theme
+  ignore_run_exports_from:
+    - {{ compiler('cxx') }}
+  missing_dso_whitelist:
+    - '*/libX11.so*'  # [linux]
```

Comment:
Is there a reason to not include the conda-forge x11 packages as dependencies?

Reply:
I never tried doing that. Are their counterparts for all these cdt's?

Reply:
For reference, the origin of this is https://github.com/conda-forge/staged-recipes/pull/10979/commits/c9a7509d84b39c5373946c8de78639215ed16ea1 from when `gtk3` went through staged-recipes.

The conda-forge x11 vs CDT discussion has come up a few times, notably https://github.com/conda-forge/gtk3-feedstock/issues/7 and https://github.com/conda-forge/cfep/pull/7. We ultimately went with CDTs for `gtk3` for now because that's what `qt` was doing. It has certainly not been headache-free, but I can't speak to potential drawbacks of depending on the conda-forge x11 packages.